src/test/groovy/com/github/gradle/node/npm/task/NpmTask_integTest.groovy

```diff
@@ -64,7 +65,8 @@ class NpmTask_integTest
         result1.task(":npmSetup").outcome == TaskOutcome.SKIPPED
         result1.task(":npmInstall").outcome == TaskOutcome.SUCCESS
         result1.task(":env").outcome == TaskOutcome.SUCCESS
-        result1.output.contains("PATH=")
+        // Sometimes the PATH variable is not defined in Windows Powershell, but the PATHEXT is
+        Pattern.compile("^PATH(?:EXT)?=.+\$", Pattern.MULTILINE).matcher(result1.output).find()
```

Comment:
Aren't we the ones setting `PATH`?

Reply:
Yes, we set the `PATH` environment variable when launching processes such as npm in this file.

But it sounds like npm does not transmit to the executed command the whole environment. I did not understand why the PATH variable was not defined and discovered that only one variable is defined, `PATHEXT` which seems to contain the extensions to try when looking for a command (for instance `.exe`, `.cmd`, `.bat`). The fact there is only one environment variable which is defined lets me think that Node.js or npm is filtering which environment variables are transmitted to the Node.js script, but I did not find any clue about that.

This broken test appeared after upgrading the default Node.js version to 12. It is probably related to an internal change. This problem appears only on Windows.

What is important for us is that additional environment variables are transmitted to the process and the test that verifies the `CUSTOM` environment variable is available in the Node.js script still works, this is the most important here.

I used the `PATH` environment variable because it is always defined in all systems, but this is no longer the case.